src/SigasiProjectCreator/DotF/DotFfileParser.py

```diff
@@ -7,163 +7,159 @@
 # 3 : input file not found
 
 import os
-import sys
 import glob
+import pathlib
 import re
 
-from SigasiProjectCreator.ArgsAndFileParser import ArgsAndFileParser
 from .parseFile import parse_dotf
-from ..convertDotFtoCsv import rebase_file
+from .. import ProjectOptions
 
 
-def abspath(path):
+def is_absolute_path(path):
+    # Check for an absolute pth on Linux or Windows, or a path which starts with an environment variable
     s_path = str(path)
-    if s_path.startswith('\\') or s_path.startswith('/') or s_path[1] == ':' or s_path.startswith('$'):
-        # this is an absolute path in Linux or Windows
+    return s_path.startswith('\\') or s_path.startswith('/') or s_path[1] == ':' or s_path.startswith('$')
+
+
+def absolute_path(path):
+    if is_absolute_path(path):
+        return path
+    return pathlib.Path(path).absolute()
+
+
+def resolve_path(path: pathlib.Path):
+    s_path = str(path)
+    # Don't resolve if it's an absolute Windows path and we're not on Windows
+    #    or if the path starts with a variable
+    if (os.name != 'nt' and s_path[1] == ':') or s_path.startswith('$'):
         return path
-    return os.path.abspath(path)
+    return path.resolve()
 
 
-def expandvars_plus(s):
-    return os.path.expandvars(re.sub(r'\$\((.*)\)', r'${\1}', s))
+def expandvars_plus(s) -> pathlib.Path:
+    return pathlib.Path(os.path.expandvars(re.sub(r'\$\((.*)\)', r'${\1}', str(s))))
 
 
 class DotFfileParser:
 
-    def __init__(self, filename):
+    def __init__(self, filename, options):
 
         self.library_mapping = dict()
         self.includes = set()
         self.defines = []
-        self.filename = ""
         self.dotfdir = ""
-        self.dotfname = ""
-        self.csvfname = ""
-        self.filecontent = []
+        self.file_content = []
         self.linked_file_mapping = dict()
 
-        self.filename = filename
-        if not os.path.isfile(filename):
-            print("*ERROR* File " + filename + " does not exist")
-            sys.exit(1)
-        if os.path.isabs(filename):
-            filename = os.path.relpath(filename)
+        assert pathlib.Path(filename).is_file(), f'*ERROR* File {filename} does not exist'
```

Comment:
Don't use asserts for user facing errors. Create a function like this somewhere and use that instead:

```python
import sys
def abort(message, code=1):
    print(message, file=sys.stderr)
    sys.exit(code)
```

An assertion results in something ugly with a stacktrace like this:

```
 test.py                                                                                                                                                                                                                                                                    ; « 11% « 4, 20
Traceback (most recent call last):
  File "/tmp/test.py", line 4, in <module>
    assert False, "hoi"
AssertionError: hoi
```

Reply:
All asserts but one (which is really internal) replaced.